pkg/rewards/claim.go

```diff
@@ -159,15 +232,69 @@ func Claim(cCtx *cli.Context, p utils.Prompter) error {
 		TokenLeaves:     convertClaimTokenLeaves(claim.TokenLeaves),
 	}
 
-	logger.Info("Validating claim proof...")
+	logger.Infof("Validating claim proof for earner %s...", earnerAddress)
 	ok, err := elReader.CheckClaim(ctx, elClaim)
 	if err != nil {
-		return err
+		return nil, nil, nil, err
 	}
 	if !ok {
-		return errors.New("failed to validate claim")
+		return nil, nil, nil, errors.New("failed to validate claim")
+	}
+	logger.Infof("Claim proof for earner %s validated successfully", earnerAddress)
+
+	return &elClaim, claim, accounts, nil
+}
+
+func Claim(cCtx *cli.Context, p utils.Prompter) error {
```

Comment:
This is more a meta point, no expecting action on this PR specifically.

@shrimalmadhur @gpabst can we start establishing a culture of breaking apart these methods?

Reply:
Sounds good to me, this PR adds `generateClaimPayload` which makes the original single Claim method a little smaller. I am also going to now try to extract the broadcast, claim output logic into a separate function.

Reply:
Sounds good. The more we put this off, the more cumbersome it will get when we are forced to do it.

Reply:
yea agree!! smaller methods. as I said before everyone should keep improving code quality and maintainability. Sometimes we do have to cut corners to ship fast but every subsequent PR should try to improve a bit.

Reply:
ok will update this one with a start to the refactoring since can reuse a good chunk of this in batch claim

Reply:
> I would recommend - feel free to just send a PR to refactor if you feel it can be improved. @bdchatham @gpabst

I agree with this ethos and would recommend not refactoring just for the sake of refactoring. If you're in an area of code that you have a chance to improve, let's do that, but I wouldn't say we're at a point of cutting refactoring tasks over finding product market fit.

My main motivator for my original comment was to simply not create these types of methods as we implement new features.